core/src/main/java/org/jruby/truffle/nodes/core/BignumNodes.java

```diff
@@ -603,6 +576,75 @@ public Object leftShift(RubyBignum a, int b) {
 
     }
 
+    @CoreMethod(names = "abs")
+    public abstract static class AbsNode extends BignumCoreMethodNode {
+
+        public AbsNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public AbsNode(AbsNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public RubyBignum abs(RubyBignum value) {
+            return value.abs();
+        }
+
+    }
+
+    @CoreMethod(names = "even?")
+    public abstract static class EvenNode extends BignumCoreMethodNode {
+
+        public EvenNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public EvenNode(EvenNode prev) {
+            super(prev);
+        }
+
+        @CompilerDirectives.TruffleBoundary
+        @Specialization
+        public boolean even(RubyBignum value) {
+            return value.bigIntegerValue().getLowestSetBit() != 0;
```

Comment:
BigInteger is not highly optimised, so this is probably the least of our concerns with the performance of Bignum. It is cached, however, as it's used internally for many other operations.